scripts/gen/replay.py

```diff
@@ -0,0 +1,61 @@
+import json
+import io
+import sys
+from typing import Union, TextIO
+
+# Sample JSON input
+sample_history = [
+  {
+    "in": "move",
+    "tick": 84
+  },
+  {
+    "in": "turn right",
+    "tick": 148
+  }
+]
+
+def process(history: list[dict]) -> list[str]:
+    # Generate output with wait commands
+    output = []
+    previous_tick = 0
+
+    for cmd in history:
+        if 'in' not in cmd:
+            continue  # output or error
+        wait_time = cmd["tick"] - previous_tick
+        if wait_time > 0:
+            output.append(f"wait {wait_time};")
+        output.append(cmd["in"] + ";")
+        previous_tick = cmd["tick"]
+    
+    return output
+    
+
+def parse(input: Union[str|TextIO]) -> list[dict]:
+    if isinstance(input, io.TextIOBase):  # file-like object
+        return json.load(input)
+    elif isinstance(input, str):  # JSON string
+        return json.loads(input)
+    else:
+        raise Exception(f"Can not parse '{type(input)}'")
+
+
+if __name__ == "__main__":
+    import argparse
+
+    parser = argparse.ArgumentParser(description="Process command JSON with wait times.")
```

Comment:
I have started to work on a REPL mode that will replay the input at proper tick, but it's going to take time and this might be useful to get some save-like functionality until then.